### DISCUSSION TITLE
Rebuild on save .cls file

### DISCUSSION THREAD

**Discussion by Soneji** [James-Yu/LaTeX-Workshop#3142] [Q&A]
Hey, this is a really nice project and I've been using it for years and really enjoy using it. I have a question, would it be possible to have LaTeX Workshop rebuild the workspace project on save of a `.cls` file? It works well when saving any of my `.tex` files but when I modify and save a `.cls` file it does not automatically rebuild the project.

Thanks so much :)
(PS: Sorry for abandoning the template, it's very long)

**Comment by tamuratak** [comment]
See 
- https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#the-root-file
- https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#the-dependencies
- https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ#saving-a-cls-file-does-not-trigger-auto-build

> The .fls files LaTeX compilers when called with the -recorder option produce a file with .fls extension containing all the files input and output during compilation. The list of input files contains all classes, packages, fonts, input .tex files, listings, graphs, ... Using latexmk always produces a .fls file.

Please fill out the [issue template](https://github.com/James-Yu/LaTeX-Workshop/blob/master/.github/ISSUE_TEMPLATE/bug-report.md) next time.
